packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/utils/cleanData.js

```diff
@@ -17,6 +17,12 @@ const cleanData = (retrievedData, currentSchema, componentsSchema) => {
       let cleanedData;
 
       switch (attrType) {
+        case 'uid':
+        case 'email':
+        case 'text':
+        case 'string':
+          cleanedData = value || null;
```

Comment:
Could `value` be ever an integer at this point? I'm asking, because it might be safer to use `??` because: `0 || null = null`, but `0 ?? null = 0`.

Could you also create/ extend the tests for `cleanData()` to reflect that change?

Reply:
Good question, indeed I didn't think about the 0 possibilities. My issue with `??` is that all idea of this fix is to prevent empty strings to be sent to the back-end, but `"" ?? null` would be "" right?
Maybe we should be more specific only preventing empty string then?

Reply:
I checked, actually if I pass the value `0` we get a string `"0"` that's why it doesn't break 🤔

Reply:
I think we need a reset field instead of this hack, what if the user really wants an empty string?

Reply:
you mean a user could want empty string as a unique value that can't be shared in 2 entries?

Reply:
@soupette @ronronscelestes and I discussed it briefly: what do you think about moving the normalization into https://github.com/strapi/strapi/blob/e58b983b6f22ce237613d93858abf194bf960a9b/packages/core/helper-plugin/lib/src/components/GenericInput/index.js#L337

We are already doing some `null`, `undefined` or `''` normalization here and could do the same for `onChange` handlers on `TextInput`s, if the value is an empty string ...

In the long-term we should clarify, what our notion of empty in the front actually is: right now it is a pretty wild mix of everything ...